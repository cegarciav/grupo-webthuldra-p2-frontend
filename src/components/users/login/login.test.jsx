import React from 'react';
import PropTypes from 'prop-types';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import MainRouter from '../../../routers/index';

function TestRouter({ path }) {
  return (
    <MemoryRouter initialEntries={[path]}>
      <MainRouter />
    </MemoryRouter>
  );
}

TestRouter.propTypes = {
  path: PropTypes.string.isRequired,
};

describe('Login view', () => {
  describe('When user is not logged-in', () => {
    it('renders the login form', async () => {
      const loginRenderer = await renderer.create(<TestRouter path="/login" />);
      const tree = loginRenderer.toJSON();
      expect(tree).toMatchSnapshot();
    });

    afterEach(() => {
      cleanup();
      jest.resetAllMocks();
    });
  });
});
