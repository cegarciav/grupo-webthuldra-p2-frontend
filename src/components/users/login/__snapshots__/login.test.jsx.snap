// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login view When user is not logged-in renders the login form 1`] = `
<main>
  <form
    action="#"
    className="form-user-inputs"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <h1>
      Ingresa tus datos
    </h1>
    <section
      className="form-field-100"
    >
      Email
      <input
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="email"
        value=""
      />
    </section>
    <section
      className="form-field-100"
    >
      Contraseña
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Contraseña"
        type="password"
        value=""
      />
    </section>
    <section
      className="form-submit"
    >
      <button
        className="button-disabled"
        disabled={true}
        type="submit"
      >
        Iniciar sesión
      </button>
    </section>
  </form>
</main>
`;
